### PR TITLE
Basic SIL representation support for coroutines

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

File: docs/SIL.rst [hunk 1]
```diff
@@ -582,6 +582,43 @@ autorelease in the callee.
...
+values are written in the result list of a function type, prefixed by
+the ``@yields`` attribute.  A yielded value may have a convention attribute,
+taken from the set of parameter attributes and interpreted as if the yield
+site were calling back to the calling function.
+
+Currently, a coroutine may not have normal results.
+
+Coroutine functions may be used in many of the same ways as normal
+function values.  However, they cannot be called with the standard
+``apply`` or ``try_apply`` instructions.
```
Comment (eeckstein) [inline docs/SIL.rst]: @rjmccall So how would a coroutine be called?

Comment (rjmccall) [reply docs/SIL.rst]: I will add that documentation when I add those instructions. :)

Comment (eeckstein) [reply docs/SIL.rst]: ok, thanks

**Comment by rjmccall** [comment]
I forgot to bump the module version, oops.

**Pull request merged**